plugins/osdn/osdn.go

```diff
@@ -207,6 +207,11 @@ func (oi *OsdnRegistryInterface) WriteNetworkConfig(network string, subnetLength
 	if err == nil {
 		if cn.Network == network && cn.HostSubnetLength == int(subnetLength) && cn.ServiceNetwork == serviceNetwork {
 			return nil
+		} else if cn.Network == network && cn.HostSubnetLength == int(subnetLength) && cn.ServiceNetwork == "" {
+			// Upgrade from 3.0.0
+			cn.ServiceNetwork = serviceNetwork
+			_, err = oi.oClient.ClusterNetwork().Update(cn)
```

Comment:
@pravisankar feel free to open a follow up issue for moving to conversion if we can figure out how